### PR TITLE
PZSP2-49 - Front page news

### PR DESCRIPTION
**Pull request by tedtheripper**

### CODE REVIEW

File: services/frontend/src/views/admin/News.vue [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<template>
+  <div>
+    <h1>Aktualności</h1>
+    <v-divider />
+    <NewsData v-bind:newsId="this.$route.params.id" />
```
Comment (noFrostoo) [inline services/frontend/src/views/admin/News.vue:5]: if im not mistaken `newsData` doesn't use  newsID

File: services/frontend/src/components/UpdateNews.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    this.getSingleNewsData();
+  },
+  methods: {
+    async getSingleNewsData() {
+      try {
+        const news = await api.getNewsById(this.newsId);
+        this.title = news.data.title;
+        this.description = news.data.description;
+      } catch (e) {
+        this.$store.dispatch("actionCheckApiError", e);
```
Comment (noFrostoo) [inline services/frontend/src/components/UpdateNews.vue:60]: If you doing that, there should be check if user is still logged in, and redirect if isn't. I think that this is not needed, but i'd leave it

File: services/frontend/src/components/NewsData.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    },
+    async removeNews(id) {
+      const choice = window.confirm(
+        "Na pewno chcesz usunąć wybraną aktualność?"
+      );
+      if (!choice) return;
+      try {
+        await api.removeNewsById(this.$store.getters["token"], id);
+      } catch (e) {
+        this.$store.dispatch("actionCheckApiError", e);
```
Comment (noFrostoo) [inline services/frontend/src/components/NewsData.vue:64]: ditto from below :)

File: services/frontend/src/components/NewsData.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    };
+  },
+  methods: {
+    async getNewsData() {
+      try {
+        const news = await api.getNews({});
+        console.log(news.data);
+        this.news = news.data;
+      } catch (e) {
+        this.$store.dispatch("actionCheckApiError", e);
```
Comment (noFrostoo) [inline services/frontend/src/components/NewsData.vue:48]: ditto from below :)

File: services/frontend/src/components/NewsData.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  </div>
+</template>
+
+<script>
+import { formatDate } from "../utils";
+import { api } from "@/api";
+
+export default {
+  name: "newsResultTable",
+  props: ["loading"],
```
Comment (noFrostoo) [inline services/frontend/src/components/NewsData.vue:35]: this is not used in the view, and you are getting data in this component , so this is redundant, in my opinion

**Review by noFrostoo — CHANGES_REQUESTED** [review]

File: services/backend/alembic/versions/97b9e398ea57_added_news_table.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    op.create_index(op.f("ix_news_title"), "news", ["title"], unique=False)
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index(op.f("ix_news_title"), table_name="news")
+    op.drop_index(op.f("ix_news_id"), table_name="news")
+    op.drop_table("news")
+    # ### end Alembic commands ###
```
[the remark this answers is not in the document]
Comment (tedtheripper) [reply services/backend/alembic/versions/97b9e398ea57_added_news_table.py:39]: yup

File: services/backend/app/api/endpoints/front_news.py [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+from app.api import deps
+from app import crud, models, schemas
+
+from fastapi import APIRouter, Depends, HTTPException
+
+router = APIRouter()
+
+
+@router.post("/")
+async def add_frontpage_news(
```
[the remark this answers is not in the document]
Comment (tedtheripper) [reply services/backend/app/api/endpoints/front_news.py]: you're right, renamed in the latest version

File: services/backend/app/api/endpoints/front_news.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+from app import crud, models
+from app.api import deps
+from app import crud, models, schemas
+
+from fastapi import APIRouter, Depends, HTTPException
+
+router = APIRouter()
+
+
+@router.post("/")
```
[the remark this answers is not in the document]
Comment (tedtheripper) [reply services/backend/app/api/endpoints/front_news.py]: added

File: services/frontend/src/views/admin/News.vue (hunk 1 above)
Comment (tedtheripper) [reply services/frontend/src/views/admin/News.vue]: You're right, prolly forgot to remove it

**Review by noFrostoo — APPROVED** [review]

**Pull request merged**